caustic/lenses/nfw.py

```diff
@@ -198,115 +198,117 @@ def _h(cls, x: Tensor) -> Tensor:
+        self, x: Tensor, y: Tensor, z_s: Tensor, P: "Packed" = None
     ) -> tuple[Tensor, Tensor]:
         """
         Compute the reduced deflection angle.
 
         Args:
-            thx (Tensor): x-coordinates in the lens plane.
-            thy (Tensor): y-coordinates in the lens plane.
+            x (Tensor): x-coordinates in the lens plane.
+            y (Tensor): y-coordinates in the lens plane.
             z_s (Tensor): Redshifts of the sources.
-            x (Optional[dict[str, Any]]): Additional parameters.
+            P (Packed): Additional parameters.
 
         Returns:
             tuple[Tensor, Tensor]: The reduced deflection angles in the x and y directions.
         """
-        z_l, thx0, thy0, m, c = self.unpack(x)
+        z_l, x0, y0, m, c = self.unpack(P)
 
-        thx, thy = translate_rotate(thx, thy, thx0, thy0)
-        th = (thx**2 + thy**2).sqrt() + self.s
-        d_l = self.cosmology.angular_diameter_dist(z_l)
-        r_s = self.get_r_s(z_l, m, c, x)
+        x, y = translate_rotate(x, y, x0, y0)
+        th = (x**2 + y**2).sqrt() + self.s
+        d_l = self.cosmology.angular_diameter_distance(z_l, P)
+        scale_radius = self.get_scale_radius(z_l, m, c, P)
         xi = d_l * th * arcsec_to_rad
-        r = xi / r_s
+        r = xi / scale_radius
 
-        alpha = (
+        deflection_angle = (
             16
             * pi
             * G_over_c2
-            * self.get_rho_s(z_l, c, x)
-            * r_s**3
+            * self.get_scale_density(z_l, c, P)
+            * scale_radius**3
             * self._h(r)
             * rad_to_arcsec
             / xi
         )
 
-        ax = alpha * thx / th
-        ay = alpha * thy / th
-        return ax, ay
-
-    def alpha(
-        self, thx: Tensor, thy: Tensor, z_s: Tensor, x: Optional[dict[str, Any]] = None
-    ) -> tuple[Tensor, Tensor]:
-        """
-        Compute the deflection angle.
-
-        Args:
-            thx (Tensor): x-coordinates in the lens plane.
-            thy (Tensor): y-coordinates in the lens plane.
-            z_s (Tensor): Redshifts of the sources.
-            x (Optional[dict[str, Any]]): Additional parameters.
-
-        Returns:
-            tuple[Tensor, Tensor]: The deflection angles in the x and y directions.
-        """
-        z_l = self.unpack(x)[0]
-
-        d_s = self.cosmology.angular_diameter_dist(z_s, x)
-        d_ls = self.cosmology.angular_diameter_dist_z1z2(z_l, z_s, x)
-        ahx, ahy = self.alpha_hat(thx, thy, z_s, x)
-        return d_ls / d_s * ahx, d_ls / d_s * ahy
-
-    def kappa(
-        self, thx: Tensor, thy: Tensor, z_s: Tensor, x: Optional[dict[str, Any]] = None
+        ax = deflection_angle * x / th
+        ay = deflection_angle * y / th
+        d_s = self.cosmology.angular_diameter_distance(z_s, P)
+        d_ls = self.cosmology.angular_diameter_distance_z1z2(z_l, z_s, P)
+        return ax * d_ls / d_s, ay * d_ls / d_s
+
+    # def deflection_angle(
```

Comment:
Why is this commented?

Reply:
I was making sure I had the reduced/physical definitions right, it's good now